File: NERP.Admin.Web/ClientApp/src/app/common/geo/geoNode.ts
import { Position, Point } from './geometry';
import { Cable } from '../../entities/models/cable';
import { Helper } from '../helper';
//import { Accessory } from '../../designs/models/design';
import { Civilwork } from '../../civilworks/models/civilwork';
import { Chamber } from '../../entities/models/chamber';
import { Pole } from '../../entities/models/pole';
import { Splitter } from '../../entities/models/splitter';
import { SpliceClosure } from '../../entities/models/spliceClosure';
import { Duct } from '../../entities/models/duct';
import { ODF } from '../../entities/models/odf';
import { Accessory } from '../../accessories/models/accessory';

export class GeoNode {
    id?: string;
    levelName?: string;
    level: number;
    coordinates?: Position;
    connections?: number;
    children?: GeoNode[];
    segments?: Segment[];

    constructor(id?:string, coordinates?: Position) {

        //if (id == null || id == "")
        //    id = Helper.generateObjectId();

        this.coordinates = coordinates;
    }
}

export class Segment {
    id: string;
    start: string;
    end: string;
    coordinates: Array<Position>;
    accessories?: Accessory[];
    civilworks?: Civilwork[];
    cable?: Cable;
    chambers?: Chamber[];
    ducts?: Duct;
    odfs?: ODF[];
    poles?: Pole[];
    splitters?: Splitter[];
    spliceClosures?: SpliceClosure[];
}

File: NERP.GeoServices.API/Models/GeoNode.cs
using NERP.Common.Models;
using System.Collections.Generic;

namespace NERP.GeoServices.API.Models
{
    public class GeoNode
    {
        public string Id { get; set; }
        public int Level { get; set; }
        public string LevelName { get; set; }
        public double[] Coordinates { get; set; }
        public double Connections { get; set; }
        public List<GeoNode> Children { get; set; }
        public List<Node> Segments { get; set; }

    }

    //public class Segment
    //{
    //    public string Id { get; set; }
    //    public Point Start { get; set; }
    //    public Point End { get; set; }
    //    public string FullPath { get; set; }
    //    public List<List<double>> Coordinates { get; set; }
    //}

}
